src/runtime/semaphore_wrapper.cc

```diff
@@ -50,6 +50,10 @@ void Semaphore::Wait() {
 }
 
 bool Semaphore::TimedWait(int timeout) {
+  // sem_timedwait does not exist in Mac OS.
+#ifdef __APPLE__
+  Wait();
```

Comment:
Please add a warning message about no wait time out for MACOS.

Reply:
added.